tests/upgrade/run.sh

```diff
@@ -130,7 +130,13 @@ validate_sensor_bundle_via_upgrader() {
         "$TEST_ROOT/bin/${TEST_HOST_PLATFORM}/upgrader" \
         -kube-config kubectl \
         -local-bundle "$deploy_dir/sensor-deploy" \
-        -workflow validate-bundle
+        -workflow validate-bundle || {
+            kill "$proxy_pid" || true
+            save_junit_failure "Validate_Sensor_Bundle_Via_Upgrader" \
+                "Failed" \
+                "Check build_log"
```

Comment:
I'm not familiar with how `upgrader` logs and exits on error, but perhaps we can get error logs (maybe stderr?) and put them in the unit failure message. I think it'd be generally helpful with prow and spyglass, but once we have junit messages posted in slack, this would make it more obvious what's going on from those messages as well. If the messages aren't helpful, or it's too difficult to sift through them to get something meaningful, then this PR LGTM